src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.cs

```diff
@@ -919,7 +918,7 @@ protected override ParseOptions OptionsCore
             Encoding? encoding,
             ImmutableDictionary<string, ReportDiagnostic>? diagnosticOptions,
             CancellationToken cancellationToken)
-            => ParseText(text, options, path, encoding, diagnosticOptions, isGeneratedCode: null, cancellationToken);
+            => ParseText(SourceText.From(text, encoding, SourceHashAlgorithm.Sha1), options, path, diagnosticOptions, isGeneratedCode: null, cancellationToken);
```

Comment:
So the SourceHashAlgorithm's current default is Sha256, but why fill in the Sha1 here? Can we get a comment in the code there? (This applies to rest of file too.)

Reply:
All our APIs use SHA1 as a default algorithm. It would be breaking if we changed the default value for existing APIs. It would also be inconsistent and confusing if new APIs used SHA256 as defaults. The default here did not change, it's just explicit now.

Reply:
Note that this is already an obsolete API so adding doc comment is not that useful.

Reply:
Not sure what the comment should say - It's backward compat, obsolete API already.

Reply:
That'd be a perfectly fine comment to add, although the APIs use one default that isn't SourceHashAlgorithms.Default?

Reply:
> the APIs use one default that isn't SourceHashAlgorithms.Default? 

Correct. All our APIs use SHA1 as default. I wonder though if we could change it to `SourceHashAlgorithms.Default`. Since default value of a parameter is embedded into the call-site it wouldn't be binary breaking change to change it. It would be source-breaking change, but that's probably ok. In this case though, the default does not flow from the API parameter but it's in the method call. So to not change the behavior we need to use SHA1.

I'd say changing these default values might be possible but certainly out of scope of this PR.

Reply:
@tmat: I wish we _were_ doing it as a part of this PR because sometimes it's difficult to tell what is back compat vs. something that should be updated. Who would be broken though? The API for reading a tree's checksum is private, so it'd only be observable for somebody actually emitting the trees and looking at the resulting PDB, right?

Reply:
Yes, it'd be observable for emit only. I'm trying to keep the existing behavior intact in this PR. We can discuss changes in the defaults separately. I'm not saying we shouldn't do it but the goal of this PR is to fix Hot Reload.